### PR TITLE
fix: rspack profile outputPath unexpected

### PR DESCRIPTION
**Pull request by 9aoy**
## Summary

should not get `api.context.distPath` before config init

## Related Links

https://github.com/web-infra-dev/rsbuild/issues/2620

## Checklist

- [ ] Tests updated (or not required).
- [ ] Documentation updated (or not required).

### CODE REVIEW

File: packages/core/src/plugins/rspackProfile.ts [hunk 1]
```diff
@@ -88,6 +86,10 @@ export const pluginRspackProfile = (): RsbuildPlugin => ({
     api.onBeforeStartDevServer(onStart);
 
     api.onAfterBuild(async ({ stats }) => {
+      const profileDir = path.join(api.context.distPath, profileDirName);
+
+      const loggingFilePath = path.join(profileDir, 'logging.json');
```
Comment (chenjiahan) [inline packages/core/src/plugins/rspackProfile.ts:91]: The join calls can be merged:

```suggestion
      const loggingFilePath = path.join(api.context.distPath, profileDirName , 'logging.json');
```

File: packages/core/src/plugins/rspackProfile.ts [hunk 2]
```diff
@@ -102,6 +106,9 @@ export const pluginRspackProfile = (): RsbuildPlugin => ({
       if (enableProfileTrace) {
         rspack.experimental_cleanupGlobalTrace();
       }
+      const profileDir = path.join(api.context.distPath, profileDirName);
```
Comment (chenjiahan) [inline packages/core/src/plugins/rspackProfile.ts:109]: same

File: packages/core/src/plugins/rspackProfile.ts [hunk 3]
```diff
@@ -59,11 +56,12 @@ export const pluginRspackProfile = (): RsbuildPlugin => ({
     const enableLogging =
       RSPACK_PROFILE === 'ALL' || RSPACK_PROFILE.includes('LOGGING');
 
-    const traceFilePath = path.join(profileDir, 'trace.json');
-    const cpuProfilePath = path.join(profileDir, 'jscpuprofile.json');
-    const loggingFilePath = path.join(profileDir, 'logging.json');
-
     const onStart = () => {
+      // can't get precise api.context.distPath before config init
+      const profileDir = path.join(api.context.distPath, profileDirName);
```
Comment (chenjiahan) [inline packages/core/src/plugins/rspackProfile.ts:61]: same

File: packages/core/src/plugins/rspackProfile.ts (hunk 2 above)
Comment (9aoy) [reply packages/core/src/plugins/rspackProfile.ts:109]: `profileDir` is also used below.

**Review by chenjiahan — APPROVED** [review]

**Pull request merged**